Admin/Admin.php

```diff
@@ -923,11 +923,37 @@ public function buildDatagrid()
      * Returns the name of the parent related field, so the field can be use to set the default
      * value (ie the parent object) or to filter the object.
      *
-     * @return string the name of the parent related field
+     * @return null|string
+     *
+     * @throws \InvalidArgumentException
      */
     public function getParentAssociationMapping()
     {
-        return $this->parentAssociationMapping;
+        if (is_array($this->parentAssociationMapping) && $this->getParent()) {
+            $parent = $this->getParent()->getCode();
+            if (array_key_exists($parent, $this->parentAssociationMapping)) {
+                return $this->parentAssociationMapping[$parent];
+            } else {
+                throw new \InvalidArgumentException(sprintf(
+                    "There's no association between %s and %s.",
+                    $this->getCode(),
+                    $this->getParent()->getCode()
+                ));
+            }
+        } else {
+            return $this->parentAssociationMapping;
+        }
+    }
+
+    /**
+     * @param string $code
+     * @param string $value
+     *
+     * @return $this
+     */
+    public function addParentAssociationMapping($code, $value)
+    {
+        $this->parentAssociationMapping[$code] = $value;
```

Comment:
@core23 @greg0ire This behaviour has been here for ages. It even works with nested keys.
It is however explicitely discouraged in the [documentation](http://fr2.php.net/manual/en/language.types.array.php#language.types.array.syntax.modifying). Too much wtf factor.